ksql-rest-app/src/main/java/io/confluent/ksql/rest/util/ClusterTerminator.java

```diff
@@ -0,0 +1,161 @@
+/**
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.rest.util;
+
+import io.confluent.ksql.KsqlEngine;
+import io.confluent.ksql.ddl.commands.DdlCommand;
+import io.confluent.ksql.ddl.commands.DdlCommandExec;
+import io.confluent.ksql.ddl.commands.DropSourceCommand;
+import io.confluent.ksql.metastore.MetaStore;
+import io.confluent.ksql.metastore.StructuredDataSource;
+import io.confluent.ksql.parser.tree.AbstractStreamDropStatement;
+import io.confluent.ksql.parser.tree.DropStream;
+import io.confluent.ksql.parser.tree.DropTable;
+import io.confluent.ksql.parser.tree.QualifiedName;
+import io.confluent.ksql.rest.server.KsqlRestConfig;
+import io.confluent.ksql.rest.server.computation.Command;
+import io.confluent.ksql.rest.server.computation.CommandId;
+import io.confluent.ksql.serde.DataSource.DataSourceType;
+import io.confluent.ksql.util.ExecutorUtil;
+import io.confluent.ksql.util.KafkaTopicClient;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.KsqlException;
+import io.confluent.ksql.util.PersistentQueryMetadata;
+import io.confluent.ksql.util.QueryMetadata;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.producer.Producer;
+
+public class ClusterTerminator {
+
+  @SuppressWarnings("unchecked")
+  public void terminateCluster(
+      final KsqlConfig ksqlConfig,
+      final KsqlEngine ksqlEngine,
+      final List<String> sourcesList,
+      final boolean isKeep,
+      final Consumer<CommandId, Command> commandConsumer,
+      final Producer<CommandId, Command> commandProducer
+  ) {
+    ksqlEngine.stopAcceptingStatemens();
+    terminateCluster(ksqlEngine, sourcesList, isKeep);
+    // Delete the command topic
+    deleteCommandTopic(ksqlConfig, ksqlEngine, commandConsumer, commandProducer);
+  }
+
+  private void terminateCluster(
+      final KsqlEngine ksqlEngine,
+      final List<String> sourcesList,
+      final boolean isKeep
+  ) {
+    // Terminate all queries
```

Comment:
remove command and move code into method called `terminateAllQueries` :D

Reply:
Moved it to a new method.